File: Back/company-service/src/models/Company.js
const mongoose = require("mongoose");

const companySchema = new mongoose.Schema({
  name: { type: String, required: true, unique: true },
  address: { type: String, required: true },
  email: { type: String, required: true },
  phone: { type: String, required: true },
  ownerId: { type: String, required: true }, // Lien avec l'utilisateur Keycloak
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model("Company", companySchema);

File: Back/MicroserviceEureka/CommercialService/src/main/java/com/example/CommercialService/models/Company.java
package com.example.CommercialService.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
@Document(collection = "companies")
public class Company {
    @Id // Identifiant unique MongoDB
    private String id;

    private String name;
    private String address;
    private String email;
    private String phone;
    private Date createdAt = new Date(); // Initialisé avec la date actuelle par défaut

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }
}
